src/components/Sheet/SheetContent.tsx

```diff
@@ -0,0 +1,482 @@
+
+        this.onResize();
+
+        setTimeout(() => this.setState({inWindowResizeScope: false}), 0);
+    };
+
+    private onResize = () => {
+        const heightChanged = this.state.prevInnerContentHeight !== this.innerContentHeight;
+
+        if (!this.sheetRef.current || !this.sheetContentRef.current || !heightChanged) {
+            return;
+        }
+
+        this.sheetContentRef.current.style.transition =
+            this.state.prevInnerContentHeight > this.innerContentHeight
+                ? `height 0s ease ${this.transitionDuration}`
+                : 'none';
+
+        this.setState({prevInnerContentHeight: this.innerContentHeight});
+
+        this.sheetContentRef.current.style.height = `${this.innerContentHeight}px`;
+        this.sheetRef.current.style.transform = `translate3d(0, -${
+            this.innerContentHeight + this.sheetTopHeight
+        }px, 0)`;
+    };
+
+    private addListeners() {
+        window.addEventListener('resize', this.onResizeWindow);
+
+        if (this.sheetRef.current) {
+            const config = {subtree: true, childList: true};
+            this.observer = new MutationObserver(this.onResize);
+            this.observer.observe(this.sheetRef.current, config);
+        }
+    }
+
+    private removeListeners() {
+        window.removeEventListener('resize', this.onResizeWindow);
+
+        if (this.observer) {
+            this.observer.disconnect();
+        }
+    }
+
+    private setHash() {
+        const {id, platform, location, history} = this.props;
+
+        if (platform === Platform.BROWSER) {
+            return;
+        }
+
+        const newLocation = {...location, hash: id};
+
+        switch (platform) {
+            case Platform.IOS:
+                if (location.hash) {
+                    hashHistory.push(location.hash);
+                }
+                history.replace(newLocation);
+                break;
+            case Platform.ANDROID:
+                history.push(newLocation);
+                break;
+        }
+    }
+
+    private removeHash() {
+        const {id, platform, location, history} = this.props;
+
+        if (platform === Platform.BROWSER || location.hash !== `#${id}`) {
+            return;
+        }
+
+        switch (platform) {
+            case Platform.IOS:
+                history.replace({...location, hash: hashHistory.pop() ?? ''});
+                break;
+            case Platform.ANDROID:
+                history.goBack();
+                break;
+        }
+    }
+
+    private shouldClose(prevProps: SheetContentInnerProps) {
+        const {id, platform, location, history} = this.props;
+
+        return (
+            platform !== Platform.BROWSER &&
+            history.action === 'POP' &&
+            prevProps.location.hash !== location.hash &&
+            location.hash !== `#${id}`
+        );
+    }
+}
+
+function withRouterWrapper(Component: React.ComponentType<SheetContentInnerProps>) {
```

Comment:
Work with router/urls should be outside of the library. This component should be "dumb"